NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
+                self.__time_logger.stop_time("distance cut")
+
+            output_buffer = {}
             # loop over all stations (each station is treated independently)
-            for iSt, self._station_id in enumerate(self._station_ids):
-                t1 = time.time()
-                triggered_showers[self._station_id] = []
-                logger.debug(f"simulating station {self._station_id}")
-
-                if self._cfg['speedup']['distance_cut']:
+            for iSt, sid in enumerate(self._station_ids):
+                if self._config['speedup']['distance_cut']:
                     # perform a quick cut to reject event group completely if no shower is close enough to the station
-                    t_tmp = time.time()
-                    vertex_distances_to_station = np.linalg.norm(vertex_positions - self._station_barycenter[iSt], axis=1)
+                    vertex_distances_to_station = np.linalg.norm(vertex_positions - station_barycenter[iSt], axis=1)
                     distance_cut = self._get_distance_cut(np.sum(shower_energies)) + 100 * units.m  # 100m safety margin is added to account for extent of station around bary center.
                     if vertex_distances_to_station.min() > distance_cut:
-                        logger.debug(f"skipping station {self._station_id} because minimal distance {vertex_distances_to_station.min()/units.km:.1f}km > {distance_cut/units.km:.1f}km (shower energy = {shower_energies.max():.2g}eV) bary center of station {self._station_barycenter[iSt]}")
-                        distance_cut_time += time.time() - t_tmp
-                        iCounter += len(shower_energies)
-                        continue
-                    distance_cut_time += time.time() - t_tmp
+                        logger.debug(f"event group {event_group.get_run_number()} is too far away from station {sid}, skipping to next station")
+                        # continue
+                output_buffer[sid] = {}
+                station = NuRadioReco.framework.station.Station(sid)
+                sim_station = NuRadioReco.framework.sim_station.SimStation(sid)
+                sim_station.set_is_neutrino()  # naming not ideal, but this function defines in-ice emission (compared to in-air emission from air showers)
+                station.set_sim_station(sim_station)
+                event_group.set_station(station)
+
+                # we allow to first only simualte trigger channels. As the trigger channels might be different per station, 
+                # we need to determine the channels to simulate first per station
+                channel_ids = self._det.get_channel_ids(sid)
+                if self.__trigger_channels is not None:
+                    if isinstance(self.__trigger_channels, dict):
+                        channel_ids = self.__trigger_channels[sid]
+                    else:
+                        channel_ids = self.__trigger_channels
```

Comment:
I would add a warning here if trigger_channels is `None`. Because this is a new feature people might forget to specify it and will run everything as trigger channel. To avoid the warning (if it is intentional) one can pass `[]`.

Reply:
ok, I renamed and added a warning output to the init function.